signature/src/suite.rs

```diff
@@ -0,0 +1,35 @@
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub enum VerificationRelation {
+    AssertionMethod,
+    Authentication,
+    CapabilityInvocation,
+    CapabilityDelegation,
+}
+
+pub const PROOF_TYPE: &str = "Ed25519Signature2018";
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Ed25519Signature(pub Vec<u8>);
+
+impl AsRef<[u8]> for Ed25519Signature {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl std::fmt::Display for VerificationRelation {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            VerificationRelation::AssertionMethod => write!(f, "assertionMethod"),
```

Comment:
it's probably simpler to do an `From<VerificationRelation> for <&static str>` here and call `write!(f, self.into())`
or you could [get fancy with macros](https://github.com/mkatychev/tache/blob/master/src/lib.rs#L170-L235)!

Reply:
What is the benefit of doing that? Wouldn't we still need to do the match set in that one?